Moose docs image alt text

cmacmackin opened the pull request:
refs #29857

## Reason
Currently none of the images in the MOOSE documentation of alt-text. This makes them inaccessible for the visually impaired that rely in screen-readers and goes against the [Web Content Accessibility Guidelines](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Text_labels_and_names?utm_source=devtools&utm_medium=a11y-panel-checks-text-label#content_with_images_must_be_labeled).

## Design
The navbar extension will now include alt-text for the GitHub (or GitLab) logos it uses.

I've added support for alt-text in the media extension. By default, it will use the value of the caption attribute if that is set. If neither alt nor caption are set, it will produce a warning encouraging users to set it.

I've gone through and added alt text for all uncaptioned images in the MooseDocs (for the main repo, anyway), so there should be no warnings issued by this PR. The text for some of these could probably be better, but I was going based on my limited understanding in some cases.

## Impact
Images in the documentation will now be more accessible to those with visual impairments.

cmacmackin committed:
Add support for alt-text in included media

cmacmackin committed:
Provide alt-text for icons on navbar

cmacmackin committed:
Added alt text to images

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3012534/), step Generate coverage on de721bb wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/30946/coverage/xfem)

</details>



This comment will be updated on new commits.

cmacmackin committed:
Regolding; partially address #30946

cmacmackin committed:
Fix link describing alt text

cmacmackin committed:
Add alt text for pictures in new tutorial; partially address #30946

cticenhour commented on tutorials/shield_multiphysics/doc/content/user_workshop/problem/step10.md:
```diff
@@ -53,6 +53,7 @@ on finding issues is too long. We switch to 2D:
 !col! width=50%
 
 !media step10_mesh.png style=width:60%;
+       alt=A 2D version of the mesh, with the blocks/materials labelled.
```
```suggestion
       alt=A 2D version of the mesh, with the blocks/materials labeled.
```

cticenhour commented on tutorials/shield_multiphysics/doc/content/user_workshop/problem/step03.md:
```diff
@@ -76,6 +78,7 @@ We use the 'water_boundary_inwards' surface because the solid block is on its pr
 Summary of the boundary conditions:
 
 !media shield_multiphysics/results/step03_bcs.png
+       alt=Diagram of the domain with the type of each boundary labelled.
```
```suggestion
       alt=Diagram of the domain with the type of each boundary labeled.
```

cticenhour commented on tutorials/darcy_thermo_mech/doc/content/workshop/problem/step11.md:
```diff
@@ -56,3 +56,4 @@ cd problems
 ## Step 11: Results
 
 !media darcy_thermo_mech/step09_result.mp4
+       alt=Temperature and VonMises stress values produced by the simulation above.
```
```suggestion
       alt=Temperature and Von Mises stress values produced by the simulation above.
```

cticenhour commented on tutorials/darcy_thermo_mech/doc/content/workshop/intro/moose_multiphysics.md:
```diff
@@ -107,7 +107,9 @@ VectorPostprocessors\\
-!media darcy_thermo_mech/simulator.mp4 style=width:70%;margin-left:auto;margin-right:auto;display:block;
+!media darcy_thermo_mech/simulator.mp4
+       style=width:70%;margin-left:auto;margin-right:auto;display:block;
+       alt=Animation illustrating which parts of the MOOSE tech stack implement the various features and how they call come together under the Mammoth reactor simulator.
```
```suggestion
       alt=Animation illustrating which parts of the MOOSE tech stack implement the various features and how they all come together under the Mammoth reactor simulator.
```

cticenhour commented on tutorials/darcy_thermo_mech/doc/content/workshop/problem/step09.md:
```diff
@@ -56,3 +56,4 @@ cd problems
 ## Step 9: Results
 
 !media darcy_thermo_mech/step09_result.mp4
+       alt=Time evolution of temperature and VonMises stress from the above simulation.
```
```suggestion
       alt=Time evolution of temperature and Von Mises stress from the above simulation.
```

cticenhour commented on tutorials/darcy_thermo_mech/doc/content/workshop/problem/step10.md:
```diff
@@ -75,3 +76,4 @@ cd problems
 ## Step 10: Multi-scale Results
 
 !media darcy_thermo_mech/step10_result.mp4
+       alt=The evolution of the macro-scale temperature and the phase of the material at the micro-scale, produced by the multi-scale simulation above.
```
```suggestion
       alt=The evolution of the macro-scale temperature and the phase of the material at the micro-scale, produced by the multiscale simulation above.
```

We tend to remove the hyphen for "multi-scale" in our documentation and papers, so I'll keep that consistent here.

cticenhour reviewed (changes requested):
Thank you @cmacmackin! Finally was able to get to this and looked through all the files. Many of my suggestions are related to British --> American English adjustments, but I found a few typos we need to resolve as well.

cmacmackin committed:
Address code review
- spelling, inclusing translations to American English
- more articles and determinants in sentences
- more syle info in pics

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

cmacmackin commented:
Thanks for the proof-reading @cticenhour. I did my best to use American English, but obviously quite a few things slipped through. I've committed all of your suggestions.

moosebuild commented:
Job [Docker moose-dev-openmpi](https://civet.inl.gov/job/3069662/) on e9d3f07 : invalidated by @cmacmackin

moosebuild commented:
Job [HPC](https://civet.inl.gov/job/3069693/) on e9d3f07 : invalidated by @cmacmackin

cticenhour commented:
Thank you for the ping @GiudGiud - I had forgotten about this one. 

@cmacmackin - You have resolved all my comments, but this branch now needs to be rebased on a newer copy of the devel branch. Apologies for the delay!

cticenhour reviewed (approved):
Thank you @cmacmackin, and thank you @GiudGiud for performing the rebase!

cmacmackin commented:
It definitely worked when I was developing it, as that was how I found all the images I needed to add alttext for. If the new pictures have been added already have captions then those will have been used, which might explain the lack of warnings.

cticenhour commented:
Right - the functionality relies on a `caption` **or** an `alt`, if I remember correctly.
